src/file/filename.ts

```diff
@@ -0,0 +1,41 @@
+import mime from 'mime-types';
+import path from 'path';
+
+import { ItemType, ItemTypeUnion } from '@/item/itemType.js';
+
+// use partial of item to be usable in backend
+export const getFilenameFromItem = (item: {
+  name: string;
+  type: ItemTypeUnion;
+  mimetype?: string;
+}): string => {
+  switch (item.type) {
+    case ItemType.APP: {
+      return `${path.basename(item.name, '.app')}${'.app'}`;
+    }
```

Comment:
I'd use the brackets only when the case has logic inside.

```suggestion
    case ItemType.APP:
      return `${path.basename(item.name, '.app')}${'.app'}`;
```